src/wp-includes/general-template.php

```diff
@@ -1376,9 +1376,11 @@ function wp_title( $sep = '&raquo;', $display = true, $seplocation = '' ) {
 	// If there's a month.
 	if ( is_archive() && ! empty( $m ) ) {
 		$my_year  = substr( $m, 0, 4 );
-		$my_month = $wp_locale->get_month( substr( $m, 4, 2 ) );
+		$my_month = substr( $m, 4, 2 );
 		$my_day   = (int) substr( $m, 6, 2 );
-		$title    = $my_year . ( $my_month ? $t_sep . $my_month : '' ) . ( $my_day ? $t_sep . $my_day : '' );
+		$title    = $my_year .
+			( '' !== $my_month ? $t_sep . $wp_locale->get_month( $my_month ) : '' ) .
```

Comment:
```suggestion
			( $my_month ? $t_sep . $wp_locale->get_month( $my_month ) : '' ) .
```
`substr()` returns `false` in PHP < 8.0, so to cover `''` and `false`, you can just use the same check that's done for `$my_day` on the line below.